### DISCUSSION TITLE
Virtual Display error

### DISCUSSION THREAD

**Discussion by chak10** [rustdesk/rustdesk#10397] [General]
### Bug Description

Failed to plug in virtual display: Failed to call SetupDiEnumDeviceInterfaces, Os { code: 259, kind: Uncategorized, message:  "Dati disponibili esauriti." }
![Screenshot (1)](https://github.com/user-attachments/assets/ce5630ba-699f-4d16-aa45-f55dfcbadb23)

### How to Reproduce

I don t know

### Expected Behavior

Work?!

### Operating system(s) on local (controlling) side and remote (controlled) side

Windows 11 -> Windows 11

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.6 -> 1.3.6

### Screenshots

![Screenshot (1)](https://github.com/user-attachments/assets/bb12ac20-c9f3-4895-9032-6c41fee41125)

**Comment by fufesou** [comment]
Could you please check the device manager? Maybe the drive is not properly installed.

![image](https://github.com/user-attachments/assets/7824ed6d-d118-4f7a-82cd-8d0e8aaeae09)

**Comment by fufesou** [comment]
You can also try uninstall the driver if possible. RustDesk will install it automatically when you use the virtual display.

![image](https://github.com/user-attachments/assets/9a6d43dc-e686-40a1-b0e0-29a853a1abf8)

**Comment by chak10** [comment]
![Screenshot 2025-01-10 112423](https://github.com/user-attachments/assets/a9df2639-540c-40e0-8d6c-c3271498424d)
I tried, but there are quite a few. I uninstalled them all, then launched RustDesk, and it creates them again, giving me the same error.

**Reply by fufesou** [reply]
@chak10 Sorry for late. I missed this reply.

Please double click the driver to show more detailed errors.

Could you please test if you can install this virtual display?  https://github.com/VirtualDisplay/Virtual-Display-Driver

**Reply by fufesou** [reply]
Please also test whether you can directly install the id currently used by RustDesk.

https://github.com/rustdesk-org/rdev/releases/download/usbmmidd_v2/usbmmidd_v2.zip

**Reply by chak10** [reply]
the error was :  - System 

  - Provider 

   [ Name]  Microsoft-Windows-Kernel-PnP 
   [ Guid]  {9c205a39-1250-487d-abd7-e831c6290539} 
 
   EventID 411 
 
   Version 0 
 
   Level 2 
 
   Task 0 
 
   Opcode 0 
 
   Keywords 0x4000000000000000 
 
  - TimeCreated 

   [ SystemTime]  2024-12-10T09:36:42.2771015Z 
 
   EventRecordID 528 
 
   Correlation 
 
  - Execution 

   [ ProcessID]  4 
   [ ThreadID]  9252 
 
   Channel Microsoft-Windows-Kernel-PnP/Configuration 
 
   Computer Server-Francesco 
 
  - Security 

   [ UserID]  S-1-5-18 
 

- EventData 

  DeviceInstanceId ROOT\DISPLAY\0000 
  DriverName rdpidd.inf 
  ClassGuid {4d36e968-e325-11ce-bfc1-08002be10318} 
  ServiceName WUDFRd 
  LowerFilters  
  UpperFilters IndirectKmd 
  Problem 0x1f 
  Status 0xc0000001

**Reply by chak10** [reply]
Unable to initialize the device driver for this hardware. (Code 37)  

Unable to find the object name.

**Reply by fufesou** [reply]
Thanks. There seems to be a compatibility issue preventing the driver from installing correctly.

Could you please try the following virtual displays?
1. https://github.com/rustdesk-org/rdev/releases/download/usbmmidd_v2/usbmmidd_v2.zip
2. https://github.com/VirtualDisplay/Virtual-Display-Driver
